[StableMIR] A few fixes to pretty printing

celinval opened the pull request:
Improve identation, and a few other rvalue printing

try-job: x86_64-msvc
try-job: test-various

celinval committed:
[StableMIR] A few fixes to pretty printing

Improve identation, and a few other rvalue printing

compiler-errors commented on tests/ui/stable-mir-print/operands.rs:
```diff
@@ -0,0 +1,25 @@
+//@ compile-flags: -Z unpretty=stable-mir --crate-type lib -C panic=abort
```
Can you please add some examples for more aggregates. Ideas that come to mind are:
1. Structs w braced args {}
2. Structs w no args
3. Closures that capture values
4. Arrays constructed per-element (`[1, 2, 3]`) style

celinval replied:
BTW, for 1, this will still print `Struct ()`. But I can add more tests

compiler-errors replied:
That's fine, I just want to see how the existing behavior is exercised. Better to have tests that are wrong so we can see when they become right, rather than have no tests at all for things we explicitly know act weirdly...

compiler-errors reviewed (approved):
LGTM, but this doesn't actually exercise most of the new code it adds, does it?

compiler-errors commented:
@rustbot author

celinval commented:
@rustbot ready

compiler-errors commented on compiler/stable_mir/src/mir/pretty.rs:
```diff
@@ -24,10 +24,12 @@ impl Debug for Place {
-        .enumerate()
-        .try_for_each(|(index, local)| write!(writer, "_{}: {}", index + 1, local.ty))?;
+    let mut sep = "";
+    body.arg_locals().iter().enumerate().try_for_each(|(index, local)| {
```
frankly i dont know why this isn't just written with a `for` loop -- currently this looks significantly worse.

celinval replied:
Do you want me to change it here?

compiler-errors replied:
Nah, it's not worth the churn

compiler-errors commented:
LGTM

@bors r+ rollup

bors commented:
:pushpin: Commit 3b2c9065f20a17c57a5eb5119652508131111da5 has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors 3b2c9065f20a17c57a5eb5119652508131111da5 -->
<!-- homu: {"type":"Approved","sha":"3b2c9065f20a17c57a5eb5119652508131111da5","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:pushpin: Commit a38d2fe599dd1ab5987f00c7f429ab2210556fe2 has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors a38d2fe599dd1ab5987f00c7f429ab2210556fe2 -->
<!-- homu: {"type":"Approved","sha":"a38d2fe599dd1ab5987f00c7f429ab2210556fe2","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

workingjubilee commented:
@bors r-

workingjubilee commented:
Failed in https://github.com/rust-lang/rust/pull/132288#issuecomment-2442917833

```diff
---- [ui] tests\ui\stable-mir-print\operands.rs stdout ----
$DIR\operands.rs
$DIR\operands.rs
\a\rust\rust\tests\ui\stable-mir-print\operands.rs
$DIR\operands.rs
Saved the actual stdout to "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\stable-mir-print\\operands\\operands.stdout"


226     debug x => _1;
227     debug z => _2;
228     bb0: {
-         _0 = {closure@Span { id: 105, repr: "$DIR/operands.rs:44:5: 44:19" }}(_1, _2);
+         _0 = {closure@Span { id: 105, repr: "C:/a/rust/rust/tests/ui/stable-mir-print/operands.rs:44:5: 44:19" }}(_1, _2);
231     }
232 }
```

compiler-errors commented:
🤔 path normalization needs to be more generic i guess???

celinval commented:
Still haven't figured out how to fix this. The test passes in my development machine (Linux) and it looks like one of the linux jobs have passed as well, which makes me wonder if this can only be reproduced in Windows

workingjubilee commented:
yeah, probably.

compiler-errors commented:
It's fascinating that the only unnormalized path is in `"the string literal"`. I would recommend looking into where compiletest does its path normalization.

celinval commented:
I tried forcing the normalization for the string case. Unfortunately, I cannot use `"` in the regex, so the normalized output is missing the opening quotes. But hopefully that will get the test passing.

celinval commented:
@rustbot ready

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit 5815815f2cafdc004f9af611740eb86fdd68e860 with merge 63c50bf89705aab19bc9828e39456c6898bc57bd...
<!-- homu: {"type":"TryBuildStarted","head_sha":"5815815f2cafdc004f9af611740eb86fdd68e860","merge_sha":"63c50bf89705aab19bc9828e39456c6898bc57bd"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11693135567/job/32567930092)
Build commit: 63c50bf89705aab19bc9828e39456c6898bc57bd (`63c50bf89705aab19bc9828e39456c6898bc57bd`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/11693135567/job/32567930092"},"merge_sha":"63c50bf89705aab19bc9828e39456c6898bc57bd"} -->

compiler-errors commented:
@bors r=compiler-errors

bors commented:
:bulb: This pull request was already approved, no need to approve it again.

- There's another pull request that is currently being tested, blocking this pull request: #132325

bors commented:
:pushpin: Commit 5815815f2cafdc004f9af611740eb86fdd68e860 has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors 5815815f2cafdc004f9af611740eb86fdd68e860 -->
<!-- homu: {"type":"Approved","sha":"5815815f2cafdc004f9af611740eb86fdd68e860","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

compiler-errors commented:
Also, I didn't approve this because I'm not satisfied with the manual normalization of paths in this test (see how I pointed out how this seems to only be necessary for paths that occur in quotes), and I wanted to look into it myself, but I've been pretty busy. It seems that you want this to land and haven't looked into it yourself, so whatever I guess.

celinval commented:
@bors r-

> Also, I didn't approve this because I'm not satisfied with the manual normalization of paths in this test (see how I pointed out how this seems to only be necessary for paths that occur in quotes), and I wanted to look into it myself, but I've been pretty busy. It seems that you want this to land and haven't looked into it yourself, so whatever I guess.

Oh, I'm really sorry about this. I didn't realize that you were going to look at it. I'm OK waiting for the proper fix.

celinval commented:
> Typo, also the @ is not needed lol

I wasn't sure about that. Do you still get notified even if I don't add the @?

compiler-errors commented:
I cannot seem to push to your PR branch since you have branch protections enabled (why?) so please force-push this to https://github.com/compiler-errors/rust/commit/3bfe7335232173e18d75da3d6b815fb2fec088ac so we can test this tweak I made.

celinval commented:
@bors try

bors commented:
:hourglass: Trying commit 3bfe7335232173e18d75da3d6b815fb2fec088ac with merge d889fd4137ec01232bd68c11eb346bb0b446199e...
<!-- homu: {"type":"TryBuildStarted","head_sha":"3bfe7335232173e18d75da3d6b815fb2fec088ac","merge_sha":"d889fd4137ec01232bd68c11eb346bb0b446199e"} -->

celinval commented:
@compiler-errors I can squash the first 3 commits of this PR if you prefer.

compiler-errors commented:
Sure. Don't push before the try build is done or else it'll cancel the run. 

If this ends up working out, then r=me; otherwise, you can revert back to the normalization status and r=me.

rust-log-analyzer commented:
The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11731368448/job/32681302795) [(plain)](https://github.com/rust-lang-ci/rust/commit/d889fd4137ec01232bd68c11eb346bb0b446199e/checks/32681302795/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  99% (49971/50310)
Updating files:  99% (50203/50310)
Updating files: 100% (50310/50310)
Updating files: 100% (50310/50310), done.
branch 'try' set up to track 'origin/try'.
Switched to a new branch 'try'
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format=%H
d889fd4137ec01232bd68c11eb346bb0b446199e
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
---- [ui] tests\ui\stable-mir-print\operands.rs stdout ----
$DIR\operands.rs
$DIR\operands.rs
\a\rust\rust\tests\ui\stable-mir-print\operands.rs
$DIR\operands.rs
Saved the actual stdout to "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\stable-mir-print\\operands\\operands.stdout"


226     debug x => _1;
227     debug z => _2;
228     bb0: {
-         _0 = {closure@Span { id: 105, repr: "$DIR/operands.rs:44:5: 44:19" }}(_1, _2);
+         _0 = {closure@Span { id: 105, repr: "C:/a/rust/rust/tests/ui/stable-mir-print/operands.rs:44:5: 44:19" }}(_1, _2);
231     }
232 }



The actual stdout differed from the expected stdout.
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args stable-mir-print\operands.rs`

error: 1 errors occurred comparing output.
status: exit code: 0
command: PATH="C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64;C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64;C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-bootstrap-tools\x86_64-pc-windows-msvc\release\deps;C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin;C:\Program Files\PowerShell\7;C:\a\_temp\msys64\mingw64\bin;C:\a\_temp\msys64\usr\local\bin;C:\a\_temp\msys64\usr\bin;C:\a\_temp\msys64\usr\bin;C:\a\rust\rust\ninja;C:\a\rust\rust\sccache;C:\a\_temp\setup-msys2;C:\Program Files\MongoDB\Server\5.0\bin;C:\aliyun-cli;C:\vcpkg;C:\Program Files (x86)\NSIS;C:\tools\zstd;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\3.1.1\x64;C:\cabal\bin;C:\ghcup\bin;C:\mingw64\bin;C:\Program Files\dotnet;C:\Program Files\MySQL\MySQL Server 8.0\bin;C:\Program Files\R\R-4.4.1\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\SeleniumWebDrivers\ChromeDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.21.13\x64\bin;C:\hostedtoolcache\windows\Python\3.9.13\x64\Scripts;C:\hostedtoolcache\windows\Python\3.9.13\x64;C:\hostedtoolcache\windows\Ruby\3.0.7\x64\bin;C:\Program Files\OpenSSL\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.432-6\x64\bin;C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\ProgramData\Chocolatey\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\dotnet;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\TortoiseSVN\bin;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files\Microsoft SQL Server\150\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\WiX Toolset v3.14\bin;C:\Program Files\Microsoft SQL Server\130\DTS\Binn;C:\Program Files\Microsoft SQL Server\140\DTS\Binn;C:\Program Files\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\Microsoft SQL Server\160\DTS\Binn;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\CMake\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.7\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps;C:\a\_temp\msys64\usr\bin\site_perl;C:\a\_temp\msys64\usr\bin\vendor_perl;C:\a\_temp\msys64\usr\bin\core_perl" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\bin\\rustc.exe" "C:\\a\\rust\\rust\\tests\\ui\\stable-mir-print\\operands.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=C:\\Users\\runneradmin\\.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=C:\\a\\rust\\rust\\vendor" "--sysroot" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2" "--target=x86_64-pc-windows-msvc" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\stable-mir-print\\operands" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\native\\rust-test-helpers" "-Z" "unpretty=stable-mir" "--crate-type" "lib" "-C" "panic=abort"
// WARNING: This is highly experimental output it's intended for stable-mir developers only.
// WARNING: This is highly experimental output it's intended for stable-mir developers only.
// If you find a bug or want to improve the output open a issue at https://github.com/rust-lang/project-stable-mir.
fn operands(_1: u8) -> () {
    let mut _0: ();
    let  _2: [u8; 10];
    let  _3: u8;
    let  _4: usize;
    let mut _5: usize;
    let mut _6: bool;
    let  _7: u8;
    let  _8: usize;
    let mut _9: (usize, bool);
    let mut _10: usize;
    let mut _11: bool;
    let mut _12: (&u8, &u8);
    let mut _13: &u8;
    let mut _14: &u8;
    let  _15: &u8;
    let  _16: &u8;
    let mut _17: bool;
    let mut _18: u8;
    let mut _19: u8;
    let  _20: core::panicking::AssertKind;
    let  _21: !;
    let mut _22: Option<Arguments<'_>>;
    let  _23: &u8;
    let  _24: u8;
    let mut _25: (&u8, &u8);
    let mut _26: &u8;
    let mut _27: &u8;
    let  _28: &u8;
    let  _29: &u8;
    let mut _30: bool;
    let mut _31: u8;
    let mut _32: u8;
    let  _33: core::panicking::AssertKind;
    let  _34: !;
    let mut _35: Option<Arguments<'_>>;
    let  _36: (u8, u8);
    let  _37: u8;
    let  _38: u8;
    let mut _39: (&u8, &u8);
    let mut _40: &u8;
    let mut _41: &u8;
    let  _42: &u8;
    let  _43: &u8;
    let mut _44: bool;
    let mut _45: u8;
    let mut _46: u8;
    let  _47: core::panicking::AssertKind;
    let  _48: !;
    let mut _49: Option<Arguments<'_>>;
    let  _50: usize;
    let mut _51: &[u8];
    let mut _52: &[u8; 10];
    let  _53: usize;
    let  _54: &usize;
    let mut _55: (&usize, &usize);
    let mut _56: &usize;
    let mut _57: &usize;
    let  _58: &usize;
    let  _59: &usize;
    let mut _60: bool;
    let mut _61: usize;
    let mut _62: usize;
    let  _63: core::panicking::AssertKind;
    let  _64: !;
    let mut _65: Option<Arguments<'_>>;
    debug val => _1;
    debug array => _2;
    debug first => _3;
    debug last => _7;
    debug left_val => _15;
    debug right_val => _16;
    debug kind => _20;
    debug reference => _23;
    debug dereferenced => _24;
    debug left_val => _28;
    debug right_val => _29;
    debug kind => _33;
    debug tuple => _36;
    debug first_again => _37;
    debug first_again_again => _38;
    debug left_val => _42;
    debug right_val => _43;
    debug kind => _47;
    debug length => _50;
    debug size_of => _53;
    debug left_val => _58;
    debug right_val => _59;
    debug kind => _63;
    bb0: {
        _2 = [_1; 10];
        _4 = 0_usize;
        _5 = 10_usize;
        _6 = Lt(_4, _5);
        assert(move _6, "index out of bounds: the length is {} but the index is {}", move _5, _4) -> [success: bb1, unwind unreachable];
    bb1: {
        _3 = _2[_4];
        _3 = _2[_4];
        _9 = CheckedSub(10_usize, 1_usize);
        assert(!move (_9.1: bool), "attempt to compute `{} - {}`, which would overflow", 10_usize, 1_usize) -> [success: bb2, unwind unreachable];
    bb2: {
    bb2: {
        _8 = move (_9.0: usize);
        _10 = 10_usize;
        _11 = Lt(_8, _10);
        assert(move _11, "index out of bounds: the length is {} but the index is {}", move _10, _8) -> [success: bb3, unwind unreachable];
    bb3: {
        _7 = _2[_8];
        _13 = &_3;
        _14 = &_7;
        _14 = &_7;
        _12 = (move _13, move _14);
        _15 = (_12.0: &u8);
        _16 = (_12.1: &u8);
        _18 = (*_15);
        _19 = (*_16);
        _17 = Eq(move _18, move _19);
        switchInt(move _17) -> [0: bb5, otherwise: bb4];
    bb4: {
        _23 = &_3;
        _24 = (*_23);
        _26 = &_24;
        _26 = &_24;
        _27 = &_3;
        _25 = (move _26, move _27);
        _28 = (_25.0: &u8);
        _29 = (_25.1: &u8);
        _31 = (*_28);
        _32 = (*_29);
        _30 = Eq(move _31, move _32);
        switchInt(move _30) -> [0: bb7, otherwise: bb6];
    bb5: {
        _20 = core::panicking::AssertKind::Eq;
        _22 = std::option::Option::None;
        _22 = std::option::Option::None;
        _21 = core::panicking::assert_failed::<u8, u8>(move _20, _15, _16, move _22) -> unwind unreachable;
    bb6: {
        _36 = (_3, _7);
        _37 = (_36.0: u8);
        _38 = (_36.0: u8);
        _38 = (_36.0: u8);
        _40 = &_37;
        _41 = &_38;
        _39 = (move _40, move _41);
        _42 = (_39.0: &u8);
        _43 = (_39.1: &u8);
        _45 = (*_42);
        _46 = (*_43);
        _44 = Eq(move _45, move _46);
        switchInt(move _44) -> [0: bb9, otherwise: bb8];
    bb7: {
        _33 = core::panicking::AssertKind::Eq;
        _35 = std::option::Option::None;
        _35 = std::option::Option::None;
        _34 = core::panicking::assert_failed::<u8, u8>(move _33, _28, _29, move _35) -> unwind unreachable;
    bb8: {
        _52 = &_2;
        _52 = &_2;
        _51 = move _52 as &[u8];
        _50 = PtrMetadata(move _51);
        _54 = &_50;
        _53 = std::mem::size_of_val::<usize>(_54) -> [return: bb10, unwind unreachable];
    bb9: {
        _47 = core::panicking::AssertKind::Eq;
        _49 = std::option::Option::None;
        _49 = std::option::Option::None;
        _48 = core::panicking::assert_failed::<u8, u8>(move _47, _42, _43, move _49) -> unwind unreachable;
    bb10: {
        _56 = &_50;
        _57 = &_53;
        _57 = &_53;
        _55 = (move _56, move _57);
        _58 = (_55.0: &usize);
        _59 = (_55.1: &usize);
        _61 = (*_58);
        _62 = (*_59);
        _60 = Eq(move _61, move _62);
        switchInt(move _60) -> [0: bb12, otherwise: bb11];
    bb11: {
        return;
    }
    bb12: {
    bb12: {
        _63 = core::panicking::AssertKind::Eq;
        _65 = std::option::Option::None;
        _64 = core::panicking::assert_failed::<usize, usize>(move _63, _58, _59, move _65) -> unwind unreachable;
}
}
fn operands::{constant#0}() -> usize {
    let mut _0: usize;
    bb0: {
        _0 = 10_usize;
    }
}
}
fn more_operands() -> [Ctors; 3] {
    let mut _0: [Ctors; 3];
    let  _1: Dummy;
    let  _2: Ctors;
    let  _3: Ctors;
    let  _4: Ctors;
    debug dummy => _1;
    debug unit => _2;
    debug struct_like => _3;
    debug tup_like => _4;
    bb0: {
        _1 = Dummy('a', core::num::<impl i32>::MIN);
        _2 = Ctors::Unit;
        _3 = Ctors::StructLike(move _1);
        _4 = Ctors::TupLike(false);
        _0 = [move _2, move _3, move _4];
    }
}
}
fn more_operands::{constant#0}() -> usize {
    let mut _0: usize;
    bb0: {
        _0 = 3_usize;
    }
}
}
fn closures(_1: bool, _2: bool) -> {closure@C:\a\rust\rust\tests\ui\stable-mir-print\operands.rs:44:5: 44:19} {
    let mut _0: {closure@C:\a\rust\rust\tests\ui\stable-mir-print\operands.rs:44:5: 44:19};
    debug x => _1;
    debug z => _2;
    bb0: {
        _0 = {closure@Span { id: 105, repr: "C:\\a\\rust\\rust\\tests\\ui\\stable-mir-print\\operands.rs:44:5: 44:19" }}(_1, _2);
    }
}
}
fn closures::{closure#0}(_1: {closure@C:\a\rust\rust\tests\ui\stable-mir-print\operands.rs:44:5: 44:19}, _2: bool) -> bool {
    let mut _0: bool;
    let mut _3: bool;
    let mut _4: bool;
    debug y => _2;
    debug x => (_1.0: bool);
    debug z => (_1.1: bool);
    bb0: {
        _4 = (_1.0: bool);
        _3 = BitXor(move _4, _2);
        switchInt(move _3) -> [0: bb2, otherwise: bb1];
    bb1: {
        _0 = true;
        goto -> bb3;
    }
---
    bb3: {
        return;
    }
}
fn Ctors::TupLike(_1: bool) -> Ctors {
    let mut _0: Ctors;
    bb0: {
        _0 = Ctors::TupLike(move _1);
    }
}
------------------------------------------
stderr: none
---
test result: FAILED. 17649 passed; 1 failed; 265 ignored; 0 measured; 2 filtered out; finished in 409.17s

Some tests failed in compiletest suite=ui mode=ui host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
Build completed unsuccessfully in 0:45:41
make: *** [Makefile:106: ci-msvc-ps1] Error 1
  network time: Thu, 07 Nov 2024 22:15:40 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11731368448/job/32684039040)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/11731368448/job/32684039040","builder_name":"checks-actions"} -->

compiler-errors commented:
There's obviously something weird going on here with us converting `\\` to `/`, and then not remapping the path correctly since that probably expects backslashes. Probably some ordering thing in the output sanitization. This also seems to do with the fact that, unlike basically everything else in the MIR output, we're using a *debug*-style output for SMIR spans.

I assume that's why this doesn't affect spans in regular MIR dumps, and if we printed the path strings outside of a quote, then perhaps compiletest would actually know what to do here.

celinval commented:
> There's obviously something weird going on here with us converting `\\` to `/`, and then not remapping the path correctly since that probably expects backslashes. Probably some ordering thing in the output sanitization. This also seems to do with the fact that, unlike basically everything else in the MIR output, we're using a _debug_-style output for SMIR spans.
> 
> I assume that's why this doesn't affect spans in regular MIR dumps, and if we printed the path strings outside of a quote, then perhaps compiletest would actually know what to do here.

Instead of normalizing the test, I could fix how StableMIR is printing the span. That won't fix the compiletest issue, but I would be able to get rid of the hacky normalization

workingjubilee commented:
cc @jieyouxu

celinval commented:
@compiler-errors I fixed how StableMIR is printing closures and coroutines so it no longer prints the span debug. With that, I removed the manual normalization from the test (and the compiletest changes). I also squashed everything into a single commit.

I created a separate issue related to the compiletest normalization.

@bors try

bors commented:
:hourglass: Trying commit dd6ddcb18e56c4e3c29e49a4a98597927d2fad35 with merge 69d16ac972b9896af6263a5073bcfca0d3d13408...
<!-- homu: {"type":"TryBuildStarted","head_sha":"dd6ddcb18e56c4e3c29e49a4a98597927d2fad35","merge_sha":"69d16ac972b9896af6263a5073bcfca0d3d13408"} -->

compiler-errors commented:
r=me if CI is green

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11734100425/job/32692856390)
Build commit: 69d16ac972b9896af6263a5073bcfca0d3d13408 (`69d16ac972b9896af6263a5073bcfca0d3d13408`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/11734100425/job/32692856390"},"merge_sha":"69d16ac972b9896af6263a5073bcfca0d3d13408"} -->

bors commented:
:pushpin: Commit dd6ddcb18e56c4e3c29e49a4a98597927d2fad35 has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors dd6ddcb18e56c4e3c29e49a4a98597927d2fad35 -->
<!-- homu: {"type":"Approved","sha":"dd6ddcb18e56c4e3c29e49a4a98597927d2fad35","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
